csclient/params/params.go

```diff
@@ -258,6 +262,30 @@ type WhoAmIResponse struct {
 	Groups []string
 }
 
+// Resource holds the metadata about a Resource defined for a charm.
+type Resource struct {
+	Name        string
+	Type        ResourceType
```

Comment:
note this is now a string since your stuff landed first.  I still think it's wrong (note the magic strings in the api conversion functions).. but I'm too tired to really care :)